spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java

```diff
@@ -62,20 +61,12 @@ public void setImportMetadata(AnnotationMetadata importMetadata) {
 		}
 	}
 
-	/**
-	 * Collect any {@link AsyncConfigurer} beans through autowiring.
-	 */
-	@Autowired(required = false)
-	void setConfigurers(Collection<AsyncConfigurer> configurers) {
-		if (CollectionUtils.isEmpty(configurers)) {
-			return;
-		}
-		if (configurers.size() > 1) {
-			throw new IllegalStateException("Only one AsyncConfigurer may exist");
-		}
-		AsyncConfigurer configurer = configurers.iterator().next();
-		this.executor = configurer::getAsyncExecutor;
-		this.exceptionHandler = configurer::getAsyncUncaughtExceptionHandler;
+	@Autowired
+	void setConfigurers(Optional<AsyncConfigurer> configurerOptional) {
```

Comment:
In my opinion, a better approach would be to create a new function which clearly shows the intent (if exists, only one configurer should be used) and deprecate the old one. This would still allow configuration using `@Primary` 
```
@Autowired(required = false)
void setAsyncConfigurer(AsyncConfigurer asyncConfigurer)
```

Reply:
I agree, that looks better comparing to version with `Optional`